la_funding_analysis/pipeline/plotters.py

```diff
@@ -0,0 +1,363 @@
+    data["Received at least one grant"] = "No"
+    data["Received at least one grant"][data["received_grant"]] = "Yes"
+    fig, ax = plt.subplots()
+    data.boxplot(factor, by="Received at least one grant")
+    plt.suptitle("")
+    plt.title(graph_title)
+    plt.ylabel(graph_ylabel)
+    plt.grid(False)
+    plt.savefig(PROJECT_DIR / "outputs/figures" / filename)
+
+
+# STRIP PLOTS (scatter plots with a discrete x axis and
+# some random jitter along the x axis)
+
+
+def set_up_strip_plot_axes(axes, ymin, ymax, graph_ylabel, graph_title, type="ngrants"):
+    """Sets up axes for the strip plots that follow.
+    Two plots are used - 'ngrants' (number of grants)
+    which plots a particular factor against the total
+    number of grants received by a LA, and 'yesno'
+    which plots the factor against whether or not
+    they specifically received the SHDDF.
+    """
+    axes.grid(axis="y", color="#d3d3d3", zorder=0)
+    if type == "ngrants":
+        # Number of grants ranges from 0 to 4 so limits extend
+        # slightly beyond this
+        xmin = -0.5
+        xmax = 4.5
+        graph_xlabel = "Number of grants"
+    if type == "yesno":
+        # Slight extension of the [0, 1] interval
+        xmin = -0.4
+        xmax = 1.4
+        graph_xlabel = "Received SHDDF"
+        axes.set_xticks([0, 1])
+        axes.set_xticklabels(["No", "Yes"])
+    axes.set_axisbelow(True)
+    axes.set_xlim([xmin, xmax])
+    axes.set_ylim([ymin, ymax])
+    axes.set_xlabel(graph_xlabel)
+    axes.set_ylabel(graph_ylabel)
+    axes.set_title(graph_title)
+    return axes
+
+
+def imd_strip_plot(data, filename):
+    """Strip plot of number of grants received by a LA
+    against the number of grants it received.
+    """
+    fig, ax = plt.subplots()
+    set_up_strip_plot_axes(
+        axes=ax,
+        ymin=10000,
+        ymax=35000,
+        graph_ylabel="IMD local concentration",
+        graph_title="IMD local concentration of local authorities\nvs number of grants obtained",
+        type="ngrants",
+    )
+    data_notna = data[~data["imd_concentration"].isna()]
+    jitter(
+        axes=ax,
+        x=data_notna["total_grants"],
+        y=data_notna["imd_concentration"],
+        alpha=0.25,
+        zorder=5,
+    )
+    plt.savefig(PROJECT_DIR / "outputs/figures" / filename)
+
+
+def fp_clusters_strip_plot(data, filename):
+    """Strip plot of LA fuel poverty against the
+    number of grants it received.
+    Certain clusters of points are coloured differently
+    for ease of reference in the report - the top left
+    cluster consists of LAs which received no grants but
+    have high fuel poverty, and the middle right cluster
+    consists of LAs which received 3 grants but have
+    relatively low fuel poverty.
+    """
+    data_notna = data[~data["fp_proportion"].isna()]
+    #
+    tl_cond1 = data_notna["total_grants"] == 0
+    tl_cond2 = data_notna["fp_proportion"] > 20
```

Comment:
nice to know how you decided on these values in the docstring

Reply:
or any number that someone could pass through a function, put them in a config file

Reply:
I've added some info on this